File: src/main/java/no/novari/instance/gateway/InstanceProcessorConfiguration.java
package no.novari.instance.gateway;

import no.novari.flyt.instance.gateway.InstanceProcessor;
import no.novari.flyt.instance.gateway.InstanceProcessorFactoryService;
import no.novari.instance.gateway.model.acos.AcosInstance;
import no.novari.instance.gateway.model.acos.AcosInstanceMetadata;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Optional;

@Configuration
public class InstanceProcessorConfiguration {

    @Bean
    public InstanceProcessor<AcosInstance> instanceProcessor(
            InstanceProcessorFactoryService instanceProcessorFactoryService,
            AcosInstanceMapper acosInstanceMapper
    ) {
        return instanceProcessorFactoryService.createInstanceProcessor(
                acosInstance -> Optional.ofNullable(acosInstance.getMetadata()).map(AcosInstanceMetadata::getFormId),
                acosInstance -> Optional.ofNullable(acosInstance.getMetadata()).map(AcosInstanceMetadata::getInstanceId),
                acosInstanceMapper
        );
    }

}

File: src/main/kotlin/no/novari/flyt/acos/instance/gateway/InstanceProcessorConfiguration.kt
package no.novari.flyt.acos.instance.gateway

import no.novari.flyt.acos.instance.gateway.model.acos.AcosInstance
import no.novari.flyt.gateway.instance.InstanceProcessor
import no.novari.flyt.gateway.instance.InstanceProcessorFactoryService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class InstanceProcessorConfiguration {
    @Bean
    fun instanceProcessor(
        instanceProcessorFactoryService: InstanceProcessorFactoryService,
        acosInstanceMapper: AcosInstanceMapper,
    ): InstanceProcessor<AcosInstance> {
        return instanceProcessorFactoryService.createInstanceProcessor(
            sourceApplicationIntegrationIdFunction = { acosInstance -> acosInstance.metadata.formId },
            sourceApplicationInstanceIdFunction = { acosInstance -> acosInstance.metadata.instanceId },
            instanceMapper = acosInstanceMapper,
        )
    }
}
